### PR TITLE
MNT: Strip PY2, fix tests, infrastructure clean-up

### LINKED ISSUE #26: Get rid of 2to3
As per https://github.com/astropy/astropy-helpers/issues/333#issuecomment-326035142 , `astropy-helpers` is dropping `2to3` support. If you don't get rid of it here, you cannot upgrade to use `astropy-helpers` for Astropy v3 or later.

### PR DESCRIPTION
**Pull request by pllim**
@jdavies-st made me look at this repo and my OCD was tingling.

Bonus: Fix #26 (xref #31)

### CODE REVIEW

**Comment by pllim** [comment]
p.s. If this works, adding an entry in https://github.com/astropy/astropy-procedures/blob/master/update-packages/helpers_3.py would enable auto-update of helpers when a new version is released.

File: .travis.yml [hunk 1]
```diff
@@ -99,10 +101,11 @@ matrix:
         - os: linux
           env: PYTHON_VERSION=3.6
                NUMPY_VERSION=1.15
+               OPENMP_EXPECTED=False
                INSTALL_CMD='python setup.py build_ext --inplace'
                CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
-               PIP_DEPENDENCIES='sphinx pytest-doctestplus pytest-openfiles'
-               TEST_CMD='pytest --open-files --doctest-rst'
+               PIP_DEPENDENCIES='sphinx pytest-openfiles'
+               TEST_CMD='pytest --open-files'
```
Comment (jdavies-st) [inline .travis.yml]: I was tempted to do this in my PR as well.  😂

Comment (pllim) [reply .travis.yml]: There is really no doc worth testing at this point, so... :woman_shrugging:

**Comment by pllim** [comment]
Oh ho, it's green!

**Pull request merged**